doc/fr/developper/writestreamconnector.rst

```diff
@@ -0,0 +1,586 @@
+
+The Centreon Broker log function should be use for log only. To write into a file, 
+we must use the LUA dedicated function. Moreover, it is possible to use parameters 
+to define the name of the log file.
+
+Edit your LUA connector:
+
+.. code-block:: lua
+
+  function init(conf)
+    logFile = conf['logFile']
+    broker_log:set_parameters(3, "/var/log/centreon-broker/debug.log")
+  end
+  
+  function write(d)
+    for k,v in pairs(d) do
+      wrintIntoFile(tostring(k .. " => " .. tostring(v)).. "\n")
+    end
+    return true
+  end
+
+  function wrintIntoFile(output)
+    local file,err = io.open(logFile, 'a')
+    if file == nil then
+      broker_log:info(3, "Couldn't open file: "..err)
+    else
+      file:write(output)
+      file:close()
+    end
+  end
+
+The **init()** function allows to get parameters and define these from Centreon 
+web interface. See technical documentation for more information.
+
+Edit also your broker output to declare this parameter:
+
+.. image:: /_static/images/developper/lua/add_parameter.png
+   :align: center
+
+Then click **Save** and go to generate the configuration and restart **cbd**.
+
+.. note::
+   Don’t forget to restart “centengine” too to create the Centreon Broker cache.
+
+Data still stored into **/var/log/centreon-broker/bbdo2file.log** log file::
+
+  name => error
+  category => 3
+  interval => 300
+  rrd_len => 3456000
+  value => 0
+  value_type => 0
+  type => 196612
+  ctime => 1522315660
+  index_id => 4880
+  element => 4
+  state => 0
+  category => 3
+  interval => 300
+  rrd_len => 3456000
+  is_for_rebuild => false
+  service_id => 1056
+  type => 196609
+  ctime => 1522315660
+  host_id => 145
+  element => 1
+  is_for_rebuild => false
+  metric_id => 11920
+
+Manipulate data
+***************
+
+Not all data are necessary. We will select only the NEB category and the events 
+regarding hosts and services status to write into a file the following data:
+
+* The type of object (HOST or SERVICE)
+* The name of host and ID
+* The description of service and ID (for service only)
+* The status and output of service
+
+To do this, we need to select not all events received by Centreon Broker but only 
+event regarding status of hosts and services objects from NEB category. By 
+following `the technical documentation <https://documentation.centreon.com/docs/centreon-broker/en/latest/dev/bbdo.html#event-categories>`_, 
+the NEB events type (real time) are linked to category 1. Moreover, the status of 
+host is link to element 14 and 24 for the status of services. To filter on these 
+events only, we will add a filter function with these properties.
+
+The new LUA script will contain:
+
+.. code-block:: lua
+
+  function init(conf)
+    logFile = conf['logFile']
+    broker_log:set_parameters(3, "/var/log/centreon-broker/debug.log")
+  end
+  
+  function write(d)
+    local output = ""
+  
+    local host_name = broker_cache:get_hostname(d.host_id)
+    if not host_name then
```

Comment:
Same kind of remark here: previously,, you wrote
if host_name == nil then
and now you write
if not host_name then

It is interesting for a tutorial to specify they are almost equivalent.

A variable X is false if it is not defined (X=nil) or false (X=false). Otherwise, it is always true, even X=0 is true.